src/ng/compile.js

```diff
@@ -814,6 +814,52 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
 
   /**
    * @ngdoc method
+   * @name $compileProvider#component
+   * @kind function
+   *
+   * @description
+   * Register a new component with the compiler.
+   *
+   * A component is a shorthand for a directive with a controller, template, and a
+   * new scope.
+   *
+   * @returns {ng.$compileProvider} Self for chaining.
+   */
+  this.component = registerComponent;
+  function registerComponent(name, bindings, controller) {
+    if (isString(name)) {
+      assertValidDirectiveName(name);
+      if (arguments.length < 3) {
+        controller = bindings;
+        bindings = controller.bindings || null;
+      }
+      return registerDirective(name, valueFn({
+        controller: controller,
+        controllerAs: controller.controllerAs || '$' + name,
```

Comment:
IMO this should be `$ctrl` or something so that it's easy to do a global find/replace when switching to Angular 2. As-is, you'd have to find/replace each directive individually.